packages/core/src/exchanges/compose.ts

```diff
@@ -1,14 +1,20 @@
-import { Exchange } from '../types';
+import { Exchange, ExchangeInput } from '../types';
 
 /** This composes an array of Exchanges into a single ExchangeIO function */
-export const composeExchanges = (exchanges: Exchange[]): Exchange => {
-  if (exchanges.length === 1) {
-    return exchanges[0];
-  }
-
-  return payload => {
-    return exchanges.reduceRight((forward, exchange) => {
-      return exchange({ client: payload.client, forward });
-    }, payload.forward);
-  };
-};
+export const composeExchanges = (exchanges: Exchange[]) => ({
+  client,
+  forward,
+}: Omit<ExchangeInput, 'dispatchDebug'>) =>
+  exchanges.reduceRight(
+    (forward, exchange) =>
+      exchange({
+        client,
+        forward,
+        dispatchDebug: e =>
+          client.debugTarget!.dispatchEvent({
```

Comment:
Will passing it like this still work with the babel plugin?

Reply:
It'll require changes but hopefully should make our babel plugin more accurate!